File: apps/inwork_mobile/lib/features/orders/presentation/widgets/order_status_timeline.dart
// OrderStatusTimeline Widget Contract
export type TimelineStepStatus = 'completed' | 'current' | 'upcoming';

export interface TimelineStep {
  title: string;
  status: TimelineStepStatus;
}

export class OrderTimelineMapper {
  static getStepsForStatus(currentStatus: string): TimelineStep[] {
    const statuses = ['Created', 'Assigned', 'Accepted', 'Started', 'Completed', 'Rated'];
    const currentIndex = statuses.indexOf(currentStatus);

    return statuses.map((step, index) => {
      let status: TimelineStepStatus = 'upcoming';
      if (index < currentIndex) {
        status = 'completed';
      } else if (index === currentIndex) {
        status = 'current';
      }
      return { title: step, status };
    });
  }
}

File: apps/inwork_mobile/lib/features/orders/presentation/widgets/order_status_timeline.ts
export type OrderTimelineStatus = 
  | 'Created' 
  | 'Assigned' 
  | 'Accepted' 
  | 'Started' 
  | 'Completed' 
  | 'Rated';

export interface TimelineStep {
  title: OrderTimelineStatus;
  status: 'completed' | 'current' | 'upcoming';
}

export class OrderTimelineMapper {
  static getStepsForStatus(currentStatus: string): TimelineStep[] {
    const flow: OrderTimelineStatus[] = [
      'Created',
      'Assigned',
      'Accepted',
      'Started',
      'Completed',
      'Rated'
    ];
    
    const currentIndex = flow.indexOf(currentStatus as OrderTimelineStatus);
    const validIndex = currentIndex === -1 ? 0 : currentIndex;

    return flow.map((item, i) => {
      let status: 'completed' | 'current' | 'upcoming' = 'upcoming';
      if (i < validIndex) {
        status = 'completed';
      } else if (i === validIndex) {
        status = 'current';
      }
      return { title: item, status };
    });
  }
}
